### ISSUE TITLE
Simple question on 13/16 Disk II firmware autodetect

### ISSUE THREAD

**Issue by Keatah** [AppleWin/AppleWin#767]
Will this be an option that can be enabled/disabled or manually selected by the user?

And what happens if there is both 13-sector and 16-sector disks present at the same time? Like say for a Copy II+ session where one is converting files from 13-sector to 16-sector disks. Or opposite that.

**Comment by tomcw** [comment]
>Will this be an option that can be enabled/disabled or manually selected by the user?

No, I didn't plan to add anything to overide this.

>And what happens if there is both 13-sector and 16-sector disks present at the same time?

The disk in drive-1 will determine the Disk II card's firmware **on a reset**.
NB. The disk image must be WOZ v2 (or higher), as this contains info about the disk image's boot sector. Any other image type in drive 1 (WOZ v1 or .dsk or .nib etc) will just revert the Disk II card's firmware to 16-sector.

So if you boot Copy II+ (which is 16-sector) eg. via Ctrl+OA+Reset, then the firmware will be 16-sector and the disk will boot.

But the actual code that Copy II+ uses to r/w the 13 or 16-sector disks will be nothing to do with the card's firmware, since the code is loaded from the Copy II+ disk. (And inserting a sector-13 disk into drive-1 won't change the firmware... unless you hit reset.)

Basically it should just work.

**Comment by tomcw** [comment]
Hi @Keatah - support for WOZ v2 13-sector images in [AppleWin 1.29.11.0](https://github.com/AppleWin/AppleWin/releases/tag/v1.29.11.0). Give it a try, and let me know if it doesn't work as expected.

**Comment by Keatah** [comment]
I think everything is working fine so far. Played a little bit with copy II+ and booting and cataloging. Good that DOS 3.3 BASICS boots 13-sector nibble images fine as before.

DOS 3.3 BASICS won't boot MicroChess WOZ2 - maybe it's supposed to be that way. Not too familiar with WOZ2 format. But of course MicroChess boots fine by itself.

So far ok. Play with it more tomorrow.

**Comment by tomcw** [comment]
>DOS 3.3 BASICS won't boot MicroChess WOZ2

btw, some info on why MicroChess doesn't work with DOS 3.3 BASICS is here: https://github.com/AppleWin/AppleWin/issues/732#issuecomment-560395441

**Comment by tomcw** [comment]
Closing issue - I can reopen if you have further questions.

**Comment by Keatah** [comment]
ABM.woz doesn't automatically boot. Need to use DOS 3.3 BASIC or BOOT13. Don't know if that's a reported issue.

**Comment by tomcw** [comment]
ABM can be found [here](https://archive.org/details/wozaday_ABM).

The WOZ image is WOZv2 format, but the `bootSectorFormat` is erroneously set to 1 (meaning 16-sector), when it should be set to 2 (meaning 13-sector).

The above URL's web-page says it was uploaded by 4am on 11th April 2020.
We just need him (or someone else) to fix this value, then re-upload the zipped image.
NB. You can't just hex edit the .woz file, as you also need to re-calc the CRC for the whole image.

**Comment by sicklittlemonkey** [comment]
I quoted you a reply to the announcement:
https://twitter.com/sicklittlemonke/status/1251144248352575493

**Comment by tomcw** [comment]
Thanks Nick for reporting, and @a2-4am for the quick fix.
@Keatah - this is now fixed on woz-a-day.
